src/index.js

```diff
@@ -232,13 +231,30 @@ class AmazonContentScript extends ContentScript {
     await this.runInWorker('click', '#nav_prefetch_yourorders')
     timeFilterSelector = await this.determineDropdownId()
     const years = await this.runInWorker('getYears', timeFilterSelector)
-    this.log('debug', 'Years :' + years)
     if (!FORCE_FETCH_ALL) {
-      // If false, we just need the first year of bills
-      years.length = 1
+      // If false, we just need the last period depending on the distanceInDays value
+      if (distanceInDays <= 30) {
+        this.log(
+          'info',
+          'lastExecution under or equals 30 days, fetching the last 30 days period'
+        )
+        years.length = 1
+        years[0] = 'last30'
```

Comment:
nit: both line can be done with : 

```javascript
years = ['last30']
```
If you change `const years` to `let years`